pandas/_libs/groupby.pyx

```diff
@@ -380,5 +380,106 @@ def group_any_all(uint8_t[:] out,
                 out[lab] = flag_val
 
 
+@cython.boundscheck(False)
+@cython.wraparound(False)
+def group_quantile(ndarray[float64_t] out,
+                   ndarray[int64_t] labels,
+                   numeric[:] values,
+                   ndarray[uint8_t] mask,
+                   float64_t q,
+                   object interpolation):
+    """
+    Calculate the quantile per group.
+
+    Parameters
+    ----------
+    out : ndarray
+        Array of aggregated values that will be written to.
+    labels : ndarray
+        Array containing the unique group labels.
+    values : ndarray
+        Array containing the values to apply the function against.
+    q : float
+        The quantile value to search for.
+
+    Notes
+    -----
+    Rather than explicitly returning a value, this function modifies the
+    provided `out` parameter.
+    """
+    cdef:
+        Py_ssize_t i, N=len(labels)
+        int64_t lab, ngroups, grp_sz, non_na_sz, grp_start=0, idx=0
+        uint8_t interp, offset
+        numeric val, next_val
+        float64_t q_idx, frac
+        ndarray[int64_t] counts, non_na_counts
+        ndarray[int64_t] sort_arr
+
+    assert <Py_ssize_t>len(values) == N
```

Comment:
This check was throwing a build warning as the lhs was returning size_t whereas `len(labels)` was returning `Py_ssize_t` (eventually replaced with N since that was duplicative).

Not sure why `len` would have returned two different types though. May be a difference between `ndarray` and `numeric`

@jbrockmendel

Reply:
A difference between ndarray and memoryview is plausible.  IIRC cython suggests using `arr.shape[0]` instead of `len(arr)` for memoryviews because the former is done wholly in C-space while the latter makes a python call.  I could imagine that being related to this, but am really just guessing.

Reply:
Good call - using shape had the same effect as the cast though seems more idiomatic. Looks like I have some windows failures to look at but will push that up along with it. Good to know for future reference as well